src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -3468,6 +3468,16 @@ internal void EnsureAnonymousTypeTemplates(CancellationToken cancellationToken)
 
         #endregion
 
+        #region Features
+
+        /// <summary>
+        /// True when the "debug-analyzers" feature flag is set.
+        /// When this flag is set, the compiler will not catch exceptions from analyzer execution to allow creating dumps.
+        /// </summary>
+        internal bool FeatureDebugAnalyzers => Feature("debug-analyzers") != null;
```

Comment:
My main nit here is letting @jaredpar name things. I would prefer a flag on the lines of `disable-analyzer-exception-handling`. `debug-analyzer` is not particularly clear what it means.

Reply:
The `debug-` prefix matches other feature flags used to debug compiler behavior. Like `debug-determinism`.

Reply:
I suppose, but we're not debugging compiler behavior here. We're disabling compiler handling of exceptions from analyzers.

Reply:
It's not so much that we're disabling handling, we're aggressingly failing fast on them.

Reply:
I do wonder if we should change the internal API name though. Seeing `DebugAnalyzers` in the code doesn't line up with the behavior as much as I would like.

Reply:
I agree, and that's part of what makes me want to change the feature flag name as well: it doesn't convey what it actually does.

Reply:
Feature flag names aren't meant to be user friendly. They're (mostly) undocumented switches we support, add and can delete at any time. I'd prefer they just follow a pattern.

Reply:
Seems like you want me to rename the property? What name do you suggest? I guess I could negate the semantics of the property and call it `CatchAnalyzerExceptions`.

Reply:
> I'd prefer they just follow a pattern.

We don't have a pattern, we have _one_ example of something named `debug-`. AFAICT, here are all the `Feature` switches we have today:

* `strict`
* `disable-length-based-switch`
* `run-nullable-analysis`
* `peverify-compat`
* `debug-determinism`
* `pdb-path-determinism`
* `enable-generator-cache`
* `nullablePublicOnly`

> Feature flag names aren't meant to be user friendly.

Sure, but we're also users of the codebase. I'd prefer to have a flag that actually conveys what it does, rather than having to read through the code to determine what it actually does if I ever come across the flag and have forgotten what it means.

Reply:
> I guess I could negate the semantics of the property and call it CatchAnalyzerExceptions.

That works for me. 

> We don't have a pattern, we have one example of something named debug-. 

How many of those are debugging compiler behavior though?

Reply:
> How many of those are debugging compiler behavior though?

I still do not think that is a relevant question, as this new switch isn't debugging compiler behavior either 🙂.

Reply:
We are ~8 comments deep on the name of a switch that will be used by a small number of customers when debugging edge cases in the compiler. We need to pick a name and get the functionality in. 

@jjonescz please make a decision.

Reply:
Alright, I will change the property name, but leave the feature flag name (because I don't know how else to name it - `disable-analyzer-exception-handling` isn't accurate; `no-catch-analyzer-exceptions` seems like an antipattern with the `no-` prefix - plus I like short name for the flag).